manifests/init.pp

```diff
@@ -57,6 +61,8 @@
   Stdlib::Absolutepath                      $package_directory         = '/opt/opensearch',
   Enum['present', 'absent']                 $package_ensure            = 'present',
   Enum['archive', 'download', 'repository'] $package_source            = 'repository',
+  Boolean                                   $pin_package               = true,
+  Integer[1,999]                            $apt_pin_priority          = 999,
```

Comment:
The priority can be greater than 999: https://manpages.debian.org/bullseye/apt/apt_preferences.5.en.html#How_APT_Interprets_Priorities

Shouldn't this default to >= 1000, since that will have the effect of ensuring exactly the specified version is installed, even if that's a downgrade?

Reply:
Done